src/equations/compressible_navier_stokes_3d.jl

```diff
@@ -0,0 +1,380 @@
+@doc raw"""
+    CompressibleNavierStokesDiffusion3D(gamma, mu, Pr, equations,
+                                        gradient_variables=GradientVariablesPrimitive())
+
+These equations contain the diffusion (i.e. parabolic) terms applied
+to mass, momenta, and total energy together with the advective terms from
+the [`CompressibleEulerEquations3D`](@ref).
+
+- `gamma`: adiabatic constant,
+- `mu`: dynamic viscosity,
+- `Pr`: Prandtl number,
+- `equations`: instance of the [`CompressibleEulerEquations3D`](@ref)
+- `gradient_variables`: which variables the gradients are taken with respect to.
+                        Defaults to `GradientVariablesPrimitive()`.
+
+Fluid properties such as the dynamic viscosity ``\mu`` can be provided in any consistent unit system, e.g.,
+[``\mu``] = kg m⁻¹ s⁻¹.
+
+The particular form of the compressible Navier-Stokes implemented is
+```math
+\frac{\partial}{\partial t}
+\begin{pmatrix}
+\rho \\ \rho \mathbf{v} \\ \rho e
+\end{pmatrix}
++
+\nabla \cdot
+\begin{pmatrix}
+ \rho \mathbf{v} \\ \rho \mathbf{v}\mathbf{v}^T + p \underline{I} \\ (\rho e + p) \mathbf{v}
+\end{pmatrix}
+=
+\nabla \cdot
+\begin{pmatrix}
+0 \\ \underline{\tau} \\ \underline{\tau}\mathbf{v} - \nabla q
+\end{pmatrix}
+```
+where the system is closed with the ideal gas assumption giving
+```math
+p = (\gamma - 1) \left( \rho e - \frac{1}{2} \rho (v_1^2+v_2^2+v_3^2) \right)
+```
+as the pressure. The terms on the right hand side of the system above
+are built from the viscous stress tensor
+```math
+\underline{\tau} = \mu \left(\nabla\mathbf{v} + \left(\nabla\mathbf{v}\right)^T\right) - \frac{2}{3} \mu \left(\nabla\cdot\mathbf{v}\right)\underline{I}
+```
+where ``\underline{I}`` is the ``3\times 3`` identity matrix and the heat flux is
+```math
+\nabla q = -\kappa\nabla\left(T\right),\quad T = \frac{p}{R\rho}
+```
+where ``T`` is the temperature and ``\kappa`` is the thermal conductivity for Fick's law.
+Under the assumption that the gas has a constant Prandtl number,
+the thermal conductivity is
+```math
+\kappa = \frac{\gamma \mu R}{(\gamma - 1)\textrm{Pr}}.
+```
+From this combination of temperature ``T`` and thermal conductivity ``\kappa`` we see
+that the gas constant `R` cancels and the heat flux becomes
+```math
+\nabla q = -\kappa\nabla\left(T\right) = -\frac{\gamma \mu}{(\gamma - 1)\textrm{Pr}}\nabla\left(\frac{p}{\rho}\right)
+```
+which is the form implemented below in the [`flux`](@ref) function.
+
+In two spatial dimensions we require gradients for three quantities, e.g.,
+primitive quantities
+```math
+\nabla v_1,\, \nabla v_2,\, \nabla v_3,\, \nabla T
+```
+or the entropy variables
+```math
+\nabla w_2,\, \nabla w_3,\, \nabla w_4\, \nabla w_5
+```
+where
+```math
+w_2 = \frac{\rho v_1}{p},\, w_3 = \frac{\rho v_2}{p},\, w_4 = \frac{\rho v_3}{p},\, w_5 = -\frac{\rho}{p}
+```
+
+#!!! warning "Experimental code"
+#    This code is experimental and may be changed or removed in any future release.
```

Comment:
```suggestion
!!! warning "Experimental code"
    This code is experimental and may be changed or removed in any future release.
```
This should render this warning properly in the docstring.